backend/src/trainees/trainees.controller.ts

```diff
@@ -99,21 +108,23 @@ export class TraineesController {
   async disable(
     @Param('id', ParseIntPipe) id: number,
   ): Promise<TraineeDisableResponseDto> {
-    try {
-      await this.traineesService.disable(id);
-    } catch (error) {
-      if (error instanceof Error && error.message === 'TRAINEE_NOT_FOUND') {
-        throw new NotFoundException('Trainee with this id does not exist.');
-      }
-      if (
-        error instanceof Error &&
-        error.message === 'TRAINEE_ALREADY_DISABLED'
-      ) {
-        throw new NotFoundException(
-          'Trainee with this id is already disabled.',
-        );
-      }
+    const disableTraineeCall = await this.traineesService.disable(id);
+
+    if (disableTraineeCall.ok) {
+      return { trainee: disableTraineeCall.data };
+    }
+
+    const { error } = disableTraineeCall;
+
+    if (error === 'TRAINEE_NOT_FOUND') {
+      throw new NotFoundException('Trainee with this id does not exist.');
+    }
+
+    if (error === 'TRAINEE_ALREADY_DISABLED') {
+      throw new MethodNotAllowedException(
```

Comment:
I'd change it to `UnprocessableEntityException`.

The `MethodNotAllowedException` would be used if you eg. sent `DELETE` to an endpoint that only supports `GET` and `POST`

> The HyperText Transfer Protocol (HTTP) **`405 Method Not Allowed`** response status code indicates that the server knows the request method, but the target resource doesn't support this method.
> 
> The server **must** generate an **`Allow`** header field in a 405 status code response. The field must contain a list of methods that the target resource currently supports.
> 
> — _[MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405)_